src/pretix/control/forms/__init__.py

```diff
@@ -220,6 +223,10 @@ def clean(self, *args, **kwargs):
             ext = ext.lower()
             if ext not in self.ext_whitelist:
                 raise forms.ValidationError(_("Filetype not allowed!"))
+
+            if ext in IMAGE_EXTS:
+                validate_uploaded_file_for_valid_image(data)
+
```

Comment:
Is `ExtValidationMixin` the _correct_ place to check for a valid image? Kind of yes, if you test, whether it can be parsed as an image (although „Ext“ kind of implies just to test whether the file-type/-extension is allowed at all). It could also go into `SizeValidationMixin` as we also check for pixel dimensions (which is also kind of size-related) and parsing the image for validity there is kind of a by-product when checkin for dimensions … not sure, what is best, just stumbled upon it. `SizeValidationMixin` seems to be used anywhere `ExtValidationMixin` is used, so should not be a problem swapping around. I am undecided though whether it needs to be swapped at all.

Reply:
Thanks for pointing that out!